packages/core/admin/admin/src/content-manager/history/pages/History.tsx

```diff
@@ -99,32 +103,35 @@ const HistoryPage = () => {
     return <Navigate to="/content-manager" />;
   }
 
-  if (isLoadingDocument || isLoadingLayout || versionsResponse.isLoading) {
+  if (isLoadingDocument || isLoadingLayout || versionsResponse.isFetching) {
     return <Page.Loading />;
   }
 
   // It was a success, handle empty data
-  if (
-    !versionsResponse.isError &&
-    (!versionsResponse.data || !versionsResponse.data?.data.length)
-  ) {
+  if (!versionsResponse.isError && !versionsResponse.data?.data?.length) {
     return <Page.NoData />;
   }
 
   // We have data, handle selected version
-  if (versionsResponse.data?.data.length && !selectedVersionId) {
+  if (versionsResponse.data?.data?.length && !selectedVersionId) {
     return (
       <Navigate
-        to={{ search: stringify({ ...query, id: versionsResponse?.data?.data[0].id }) }}
+        to={{ search: stringify({ ...query, id: versionsResponse.data.data[0].id }) }}
         replace
       />
     );
   }
 
-  const selectedVersion = versionsResponse.data?.data.find(
+  const selectedVersion = versionsResponse.data?.data?.find(
     (version) => version.id.toString() === selectedVersionId
   );
-  if (versionsResponse.isError || !layout || !schema || !selectedVersion) {
+  if (
+    versionsResponse.isError ||
+    !layout ||
+    !schema ||
+    !selectedVersion ||
+    versionsResponse.data.error
```

Comment:
This looks like the same check done a different way: `versionsResponse.isError` vs `versionsResponse.data.error` they're different?

Reply:
Yes it's exactly the same, but typescript doesn't know that, so without this extra check there's an error [on this line](https://github.com/strapi/strapi/blob/725c24ce394743f6deb70349ae7ed757da013ff5/packages/core/admin/admin/src/content-manager/history/pages/History.tsx#L151)

Reply:
That's a big time bummer and sort of frustrating.